server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java

```diff
@@ -216,21 +275,30 @@ private void createIndexWithDocsAndEnsureGreen(int numReplicasIndex, int numOfDo
         ensureGreen();
     }
 
-    private void takeSnapshot(Client client, String... indices) {
-        takeSnapshot(client, null, indices);
+    private void takeSnapshot(Client client, String snapshotName, String repoName, Boolean doesRepoExist, String... indices) {
+        takeSnapshot(client, null, snapshotName, repoName, doesRepoExist, indices);
     }
 
-    private void takeSnapshot(Client client, Settings.Builder repositorySettings, String... indices) {
+    private void takeSnapshot(
+        Client client,
+        Settings.Builder repositorySettings,
+        String snapshotName,
+        String repoName,
+        Boolean doesRepoExist,
+        String... indices
+    ) {
         logger.info("--> Create a repository");
-        if (repositorySettings == null) {
-            createRepository("test-repo", "fs");
-        } else {
-            createRepository("test-repo", "fs", repositorySettings);
+        if (!doesRepoExist) {
```

Comment:
I would decompose this into `createRepository` and `takeSnapshot` methods as opposed to using a boolean parameter like this.

Also, why parameterize the repo name? Is there any case where the name isn't "test-repo"?

Reply:
Decomposed